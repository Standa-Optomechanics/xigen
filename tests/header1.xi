protocol "v1"
defaults with crc, answer, public

command "command_move" writer "move" (18)
with inline
fields:
  int32s position
  int16s uposition
  reserved 6

command "foobar" universal "fbr" (18)
fields:
  int32s position
  int16s uposition
  reserved 6

command "get_foobarbaz" reader "gfbb" (18)
fields:
  int32s position
  int16s uposition
  reserved 6

command "macguffin" reader "gmcg" (26)
fields:
  int32u field1
  int32s field2
  int16u field3
  int16s field4
  int8u field5
  int8s field6
	int16u flag additional_flags		
	float foobar

command "left" writer "left" (4)

command "leftX" writer "left" (4)
without lock

command "serial_number" writer "sser" (14)
with inline
fields:
	int32u serial
	int32u key

command "serial_number" reader "gser" (10)
with inline
fields:
	int32u serial

command "stringified" reader "geti" (36)
fields:
	char manufacturer[4]
	char manufacturer_id[2]
	char product_description[8]
	reserved 16

command "stringifiedX" reader "geti" (10)
without lock
fields:
	int32u foobar

command "reset" writer "rest" (4)
without answer, lock

command "non_public_struct_private1" reader "get1" (10)
without public
fields:
	int32u foo
	
command "non_public_struct_private2" reader "get2" (10)
without public, publicstruct
fields:
	int32u foo		
	
command "non_public_struct_public" reader "get3" (10)
without public
with publicstruct
fields:
	int32u foo

command "inlinearray" writer "winl" (40)
with inline
fields:
	int32u arr[8] metalen
	int16u extra

/** $XIW
	this should test inline parameter 'unsigned char' type 
*/
command "inlinearray2" writer "winm" (14)
with inline
fields:
	int8u arr[8] metalen

/** $XIW
 */
/** $XIS
 * This must be changed later to produce an array of uint8_t at libximc header generation. Now it is just an unsigned int. The best way is to mark it with a special type aka 'bytes foo[8]'
*/
command "truearray" writer "trua" (14)
fields:
	int8u foo[8]

/** vim: set ft=c: */
