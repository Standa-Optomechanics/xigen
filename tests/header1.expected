
/** @cond DO_NOT_WANT */
#define LIBXIMC_PROTOCOL_VERSION 1
/** @endcond */


	typedef struct
	{
		int position;
		int uposition;
	} foobar_t;

	typedef struct
	{
		int position;
		int uposition;
	} get_foobarbaz_t;

	typedef struct
	{
		unsigned int field1;
		int field2;
		unsigned int field3;
		int field4;
		unsigned int field5;
		int field6;
		unsigned int additional_flags;
		float foobar;
	} macguffin_t;

	typedef struct
	{
		char manufacturer[5];
		char manufacturer_id[3];
		char product_description[9];
	} stringified_t;

	typedef struct
	{
		unsigned int foobar;
	} stringifiedX_t;

	typedef struct
	{
		unsigned int foo;
	} non_public_struct_public_t;

	typedef struct
	{
		unsigned int foo[8];
	} truearray_t;
	result_t XIMC_API command_move (device_t id, int position, int uposition);

	result_t XIMC_API set_foobar (device_t id, const foobar_t* foobar);

	result_t XIMC_API get_foobar (device_t id, foobar_t* foobar);

	result_t XIMC_API get_foobarbaz (device_t id, get_foobarbaz_t* the_get_foobarbaz);

	result_t XIMC_API get_macguffin (device_t id, macguffin_t* macguffin);

	result_t XIMC_API command_left (device_t id);

	result_t XIMC_API command_leftX (device_t id);

	result_t XIMC_API set_serial_number (device_t id, unsigned int serial, unsigned int key);

	result_t XIMC_API get_serial_number (device_t id, unsigned int* serial);

	result_t XIMC_API get_stringified (device_t id, stringified_t* stringified);

	result_t XIMC_API get_stringifiedX (device_t id, stringifiedX_t* stringifiedX);

	result_t XIMC_API command_reset (device_t id);

	result_t XIMC_API set_inlinearray (device_t id, const unsigned int* arr, unsigned int extra);

	result_t XIMC_API set_inlinearray2 (device_t id, const uint8_t* arr);

	result_t XIMC_API set_truearray (device_t id, const truearray_t* truearray);

